### PR TITLE
Im unable to use the delete button #7

### PR DESCRIPTION
**Pull request by jeevikasirwani**
added a handler function

### CODE REVIEW

File: src/pages/Main.tsx [hunk 1]
```diff
@@ -10,7 +10,11 @@ export default function Main() {
     make(station)
     navigate('/player')
   }
+function handler_delete(id:string){
+  // Remove the station from the store based on its ID
+  useStationsStore.method.remove(id)
```
Comment (tristonarmstrong) [inline src/pages/Main.tsx:15]: `.method` isn't a valid property on `useStationsStore`. Its `methods`. Easy mistake.

Tho, `.remove` isnt even a property on `methods` at all.. You'll need to head on over to `src/hooks/stationStores.ts` and you'll see that theres a `delete` function on there already. Tho it'll need to be modified so it correctly updates the file that stores  all of the stations.

File: src/pages/Main.tsx [hunk 2]
```diff
@@ -37,7 +41,8 @@ export default function Main() {
             <img src={s.avatar} width={40} height={40} className="w-20 rounded-xl" />
             <div className="flex flex-col gap-2 w-full">
               <h2 className="text-xl">{s.title}</h2>
-              <button className="border border-red-500 rounded w-full text-red-500 px-4">Delete</button>
+
+              <button onClick={()=>handler_delete(s.id)} className="border border-red-500 rounded w-full text-red-500 px-4">Delete</button>
```
Comment (tristonarmstrong) [inline src/pages/Main.tsx:45]: `onClick`  is not a valid attribute on button. May need to enable your LSP

**Review by tristonarmstrong — CHANGES_REQUESTED** [review]
Nice work, but needs a few changes:
- [ ] fix onclick handler to use correct attribute
- [ ] fix handler delete function to use correct properties on `useStationsStore`
- [ ] update the `useStationsStore`  store object's `delete` method to properly write to file

File: src/pages/Main.tsx (hunk 1 above)
Comment (jeevikasirwani) [reply src/pages/Main.tsx:15]: delete: (stationId) =>
      set(
        (state) => state?.filter((station) => station.id !== stationId) ?? []
      ),
      const updatedStations = useStationsStore.getState();
      try {
        await writeTextFile("stations.json", JSON.stringify(updatedStations));
      } catch (error) {
        console.error("Error writing to file:", error);
      }                                          
                                                                                                                                                                                                                 so adding a try catch block in useStationsStore delete function.





changes in main.ts      
                                                                    
   function handler_delete(id:string){
               // Remove the station from the store based on its ID
               useStationsStore.delete(id);
}

Comment (tristonarmstrong) [reply src/pages/Main.tsx:15]: If you're just asking if this code works, be sure to use backticks, so it looks like this:
```ts
console.log("hello world");
```
So i can read it comfortably, as well as you should annotate what you're intention is so i know whether im answering a question or what. Right now you just gave me some code and said something but i dont know what you're trying to communicate

**Comment by tristonarmstrong** [comment]
Closing due to slow iteration and poor communication. Will work on this myself

**Pull request closed without merging**